smtp/client.ts

```diff
@@ -1,87 +1,45 @@
-const { SMTP, state } = require('./smtp');
-const { Message, create } = require('./message');
-const addressparser = require('addressparser');
+// @ts-ignore
+import addressparser from 'addressparser';
+import { Message } from './message';
+import { SMTP, SMTPState } from './smtp';
+
+export interface MessageStack {
+	callback: (error: Error | null, message: Message) => void;
+	message: Message;
+	attachment: import('./message').MessageAttachment;
+	text: string;
+	returnPath: string;
+	from: string;
+	to: string | { address: string }[];
+	cc: string[];
+	bcc: string[];
+}
+
+export class Client {
+	public smtp: SMTP;
+	public queue: MessageStack[] = [];
+	public timer: NodeJS.Timer | null = null;
+	public sending = false;
+	public ready = false;
 
-class Client {
 	/**
-	 * @typedef {Object} MessageStack
-	 * @property {function(Error, Message): void} [callback]
-	 * @property {Message} [message]
-	 * @property {string} [returnPath]
-	 * @property {string} [from]
-	 * @property {string} [subject]
-	 * @property {string|Array} [to]
-	 * @property {Array} [cc]
-	 * @property {Array} [bcc]
-	 * @property {string} [text]
-	 * @property {*} [attachment]
-	 *
-	 * @typedef {Object} SMTPSocketOptions
-	 * @property {string} key
-	 * @property {string} ca
-	 * @property {string} cert
-	 *
-	 * @typedef {Object} SMTPOptions
-	 * @property {number} [timeout]
-	 * @property {string} [user]
-	 * @property {string} [password]
-	 * @property {string} [domain]
-	 * @property {string} [host]
-	 * @property {number} [port]
-	 * @property {boolean|SMTPSocketOptions} [ssl]
-	 * @property {boolean|SMTPSocketOptions} [tls]
-	 * @property {string[]} [authentication]
-	 * @property {function(...any): void} [logger]
-	 *
-	 * @constructor
-	 * @param {SMTPOptions} server smtp options
+	 * @param {*} server smtp options
 	 */
-	constructor(server) {
+	constructor(server: Partial<import('./smtp').SMTPOptions>) {
```

Comment:
why the pattern of importing inline like this?

Reply:
`typescript-eslint` throws "unused imports" errors. seems it's not capable of detecting type usages just yet.

Reply:
ah. could you import these at the top and do an inline comment ignore instead?

Reply:
done :)